File: app/src/main/java/com/lazycouple/restapiclient/util/ConfigProperties.java
package com.lazycouple.restapiclient.util;

import android.content.Context;

import com.lazycouple.restapiclient.R;
import com.lazycouple.restapiclient.db.model.Parameter;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * Created by noco on 2016-10-13.
 */
public class ConfigProperties {
    private final static String TAG = ConfigProperties.class.getSimpleName();

    public static List<Parameter> getApiInfo(Context context, String arrayName) {

        int id = context.getResources().getIdentifier(arrayName, "array", context.getPackageName());
        String[] array = context.getResources().getStringArray(id);

        List<Parameter> params = new ArrayList<>();
        for(String str : array)
        {
            String[] split = str.split("\\|");
            params.add(new Parameter(split[0], split[1]));
        }
        return params;
    }

    public static List<String> getHistories(Context context) {
        String[] array = context.getResources().getStringArray(R.array.history);
        List<String> histories = new ArrayList<String>(Arrays.asList(array));
        return histories;
    }

}

File: app/src/main/kotlin/com/lazycouple/restapiclient/util/ConfigProperties.kt
package com.lazycouple.restapiclient.util

import android.content.Context

import com.lazycouple.restapiclient.R
import com.lazycouple.restapiclient.repository.local.model.Parameter

import java.util.ArrayList
import java.util.Arrays

/**
 * Created by noco on 2016-10-13.
 */
object ConfigProperties {
    private val TAG = ConfigProperties::class.java.simpleName

    fun getApiInfo(context: Context, arrayName: String): List<Parameter> {

        val id = context.resources.getIdentifier(arrayName, "array", context.packageName)
        val array = context.resources.getStringArray(id)

        val params = ArrayList<Parameter>()
        for (str in array) {
            val split = str.split("\\|".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            params.add(Parameter(split[0], split[1]))
        }
        return params
    }

    fun getHistories(context: Context): List<String> {
        val array = context.resources.getStringArray(R.array.history)
        val histories = ArrayList(Arrays.asList(*array))
        return histories
    }

}
